src/konnektor/network_planners/generators/maximal_network_generator.py

```diff
@@ -128,6 +128,7 @@ def generate_ligand_network(self, components: Iterable[Component]) -> LigandNetw
                     else:
                         try:
                             best_mapping = next(mapping_generator)
+                            break
```

Comment:
I am more concerned that this _is_ considered covered by tests, but the tests did not break when this change was made.